src/MacroTools/Mechanics/StructurePacking.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using WCSharp.Events;
+using static War3Api.Common;
+
+namespace MacroTools.Mechanics
+{
+  /// <summary>
+  /// System to allow structures to convert themselves into units. Those units can then use an ability to rebuild the building.
+  /// </summary>
+  public class StructurePacking
+  {
+    /// <summary>
+    /// Object that contains information and methods for packable structures
+    /// </summary>
+    public class PackableStructure
+    {
+      private static readonly Dictionary<int, PackableStructure> PackableStructureById = new();
+      private int _packedUnitId;
+      private int _buildAbility;
+      private string? _structureModel;
```

Comment:
Rather than making this nullable let's just initialize it to something valid.
```suggestion
      private string _structureModel = string.Empty;
```